Обновить рекомендацию

s3nkyh opened the pull request.

Commit:
added: RecomContrl, RecomDto, RecomService, SkillOfferDto

Commit:
--

Commit:
-

Commit:
--

Commit:
исправил RecommendationMapper

Commit:
убрал лишние аннотации

Commit:
исправил RecommendationMapper
добавил метод deleteAllAndCreate в RecommendationService

antonykapitukov commented on src/main/java/school/faang/user_service/service/RecommendationService.java:
```diff
@@ -0,0 +1,72 @@
+    private final RecommendationRepository recommendationRepository;
+    private final SkillOfferRepository skillOfferRepository;
+    private final UserSkillGuaranteeRepository userSkillGuaranteeRepository;
+    private final RecommendationMapper mapper;
```
рекомендую назвать recommendationMapper

antonykapitukov commented on src/main/java/school/faang/user_service/mapper/RecommendationMapper.java:
```diff
@@ -0,0 +1,53 @@
+    @Mapping(target = "skillOffers", source = "skillOffersDto")
+    @Mapping(target = "request", ignore = true)
+    @Mapping(target = "updatedAt", ignore = true)
+    Recommendation toEntity(RecommendationDto recommendationDto);
```
рекомендую назвать метод toRecommendation и с остальными методами аналогично

antonykapitukov commented on src/main/java/school/faang/user_service/utils/ValidationRecommendationUtils.java:
```diff
@@ -0,0 +1,48 @@
+
+        if (period.toTotalMonths() <= 6) {
+            throw new DateTimeException("It is not possible to give a recommendation to this user, since the last" +
+                    " recommendation was made less than 6 months ago.");
```
вроде Паша говорил писать логи и сообщения на русском

antonykapitukov commented on src/main/java/school/faang/user_service/utils/ValidationRecommendationUtils.java:
```diff
@@ -0,0 +1,48 @@
+public class ValidationRecommendationUtils {
+    public static void validateRecommendationContent(RecommendationDto recommendationDto) {
+        if (recommendationDto.getContent() == null) {
+            throw new DataValidationException("The content of the recommendation can't be null");
```
рекомендую вынести все строки-сообщения как private static константы

Commit:
переименовал: mapper -> recommendationMapper,
toEntity -> toRecommendation и др.
вынес все сообщения в константы и сделал их русскими

Commit:
исправил RecommendationService,
RecommendationValidation
добавил метод в RecommendationRepository и SkillOfferRepository

Commit:
-

Commit:
Испаврил RecommendationMapper
